src/psyclone/psyad/adjoint_visitor.py

```diff
@@ -148,6 +150,36 @@ def schedule_node(self, node):
             # else:
             #     node_copy.children.append(result)
 
+        # Creating the adjoint may have altered the way variables are
+        # accessed within the code. If any of the active variables are
+        # subroutine arguments then we must update the access property of
+        # the associated ArgumentInterface.
+        # Since a piece of code could contain many Schedules, ensure we are
+        # currently handling the outermost one.
+        if not node.ancestor(Schedule):
```

Comment:
I take your point (at least, what I understand to be your point). I've changed this so that it checks whether node is an instance of Routine instead. An alternative would be to add a Routine handler to the visitor but since a Routine is a Schedule I thought that might get complicated.